test/udp_sender.js

```diff
@@ -249,4 +251,23 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', done => {
+        let spy = sinon.spy(console, 'log');
+        sender._host = 'foo.bar.com';
+        sender._port = 1234;
+        new Tracer(
+            'test-service-name',
+            new RemoteReporter(sender),
+            new ConstSampler(true)
+        ).startSpan('testSpan').finish();
+        sender.flush();
+        let oldLog = console.log;
+        console.log = message =>  {
```

Comment:
shouldn't this be done _before_ `sender.flush()`?

also, since you're resetting the function back to oldLog, I don't think you need sinon anymore.